pkg/parse/quote.go

```diff
@@ -99,26 +106,39 @@ func rtohex(r rune, w int) []byte {
 func quoteDouble(s string) string {
 	var buf bytes.Buffer
 	buf.WriteByte('"')
-	for _, r := range s {
+	for len(s) > 0 {
+		r, w := utf8.DecodeRuneInString(s)
+		if r == utf8.RuneError {
+			// An invalid UTF-8 sequence was seen -- encode first byte as a hex literal.
+			buf.WriteByte('\\')
+			buf.WriteByte('x')
+			buf.Write(rtohex(rune(s[0]), 2))
+			s = s[w:]
+			continue
+		}
+
+		// The common case -- a valid UTF-8 sequence was decoded so we have a rune.
 		if e, ok := doubleUnescape[r]; ok {
```

Comment:
Make this `else if`, and it's unnecessary to duplicate the `s = s[:w]` part in the `RuneError` branch.